platforms/software/dependency-management/src/test/groovy/org/gradle/api/internal/notations/DependencyStringNotationConverterTest.groovy

```diff
@@ -170,49 +169,8 @@ class DependencyStringNotationConverterTest extends Specification {
         !d.changing
     }
 
-    def "can create client module"() {
-        def parser = new DependencyStringNotationConverter(TestUtil.instantiatorFactory().decorateLenient(), DefaultClientModule, SimpleMapInterner.notThreadSafe());
-
-        when:
-        def d = parse(parser, 'org.gradle:gradle-core:10')
-
-        then:
-        d instanceof DefaultClientModule
-        d.name == 'gradle-core'
-        d.group == 'org.gradle'
-        d.version == '10'
-        d.versionConstraint.requiredVersion == '10'
-        d.versionConstraint.preferredVersion == ''
-        d.versionConstraint.strictVersion == ''
-        d.versionConstraint.rejectedVersions == []
-        d.transitive
-
-        !d.force
-    }
-
-    def "client module ignores the artifact only notation"() {
-        def parser = new DependencyStringNotationConverter(TestUtil.instantiatorFactory().decorateLenient(), DefaultClientModule, SimpleMapInterner.notThreadSafe());
-
-        when:
-        def d = parse(parser, 'org.gradle:gradle-core:10@jar')
-
-        then:
-        d instanceof DefaultClientModule
-        d.name == 'gradle-core'
-        d.group == 'org.gradle'
-        d.version == '10@jar'
-        d.versionConstraint.requiredVersion == '10@jar'
-        d.versionConstraint.preferredVersion == ''
-        d.versionConstraint.strictVersion == ''
-        d.versionConstraint.rejectedVersions == []
-        d.transitive
-
-        !d.force
-        d.artifacts.size() == 0
-    }
-
     def "parses short hand-notation #notation for strict dependencies"() {
-        def parser = new DependencyStringNotationConverter(TestUtil.instantiatorFactory().decorateLenient(), DefaultClientModule, SimpleMapInterner.notThreadSafe());
+        def parser = new DependencyStringNotationConverter(TestUtil.instantiatorFactory().decorateLenient(), DefaultExternalModuleDependency, SimpleMapInterner.notThreadSafe());
```

Comment:
🤔 Weird that we can just change the wanted type here and have this test still work.  Can we mock an interface instead?

Reply:
Seems like that would require more boilerplate than necessary. 

We would need to write _some_ type to pass in here, and yes we can write that in the test, but it would also need a constructor, as we try to instantiate it [here](https://github.com/gradle/gradle/blob/f3c41e134f252ead7e2e398c566ee06e209b5dc5/platforms/software/dependency-management/src/main/java/org/gradle/api/internal/notations/DependencyStringNotationConverter.java#L60-L61)

I can do it if you want but I don't really see the point. This class is written to accept any type that has a constructor with this well-known format. TBH I would rather change the way that works than create some test version of this interface

Reply:
OK, I didn't look and thought it would be easy to do.